tests/testthat/test-derive_interp_records.R

```diff
@@ -0,0 +1,19 @@
+# derive_interp_records ----
```

Comment:
please add some tests for actual interpolated values to be sure the function works as expected, as currently these tests are only error checking. see https://github.com/pharmaverse/admiralpeds/pull/54/files for example

Reply:
it's still ongoing. I'll try to complete it by tomorrow EOB.

Reply:
I have implemented them, could you please review them ? I had to add `library(admiraldev)` and `library(dplyr)` as it was failing otherwise. Could you confirm this is necessary ?
![image](https://github.com/user-attachments/assets/878f40ca-f8e5-4e36-b6ff-e9f290456d03)

Reply:
@zdz2101 left the following comment on Laura's PR if it helps:

Per [textbook](https://r-pkgs.org/testing-basics.html#test-organisation) you'll want to avoid calling libraries in unit-test suite, if you need to use {pkgname} functions just use pkgname::xxx() format, demonstrated below with the tibble::tribble() call. Although it's ideal to this with all outside-functions, I think dplyr ones are so commonly used we can use the importFrom in the admiralpeds-package.R to avoid excessive use

Also can you merge latest main and resolve the conflict with `R/admiralpeds-package.R`

Reply:
Implemented. Thanks for the guidance.